### DISCUSSION TITLE
Data Transformation - Bagaimana menentukan apakah data sebaiknya diubah dengan log? sqrt? sqr?

### DISCUSSION THREAD

**Discussion by Hertzspr** [teamalgoritma/community#49] [Q&A] [answered]
Bagaimana menentukan apakah data sebaiknya diubah dengan log? sqrt? sqr? Apa saja pertimbangannya?

Misalkan, saya punya data cumulative berikut:
Garis merah, adalah nilai yang rencanaanya akan dijadikan Dependent Variable dalam suatu prediksi.
Sedangkan data lain terlihat terlalu melenceng jauh dari data garis merah tersebut.
![data awal](https://user-images.githubusercontent.com/95233522/163726644-ff191d40-6968-4fb5-83e7-a2f6f93b8a81.png)

-
Kemudian saya coba ubah data lain (biru, hijau, hitam) dengan log()
![data log transformed](https://user-images.githubusercontent.com/95233522/163726670-d29bae4c-2f9e-4292-a2cb-2eae23ed6b9e.png)

Tapi saya penasaran dengan bentuk data lain, saya coba dengan sqrt(). Sepertinya lebih selaras yang log ya daripada sqrt?
![data square root transformation 000010](https://user-images.githubusercontent.com/95233522/163726688-fb48b5de-8bf5-4a39-b045-82378e740321.png)

Nah, intinya saya mohon arahan, atau mungkin rujukan bacaan untuk mencerahkan perihal data transformation ini? Di industri best practicenya seperti apa? Pertimbangannya seperti apa? Terima kasih.

**Comment by inytss** [comment, marked as answer]
Dear @Hertzspr , 
Dalam menentukan metode transformasi data yang akan digunakan terutama dalam pemodelan machine learning secara umum dapat dilakukan trial dan error dikarenakan tidak ada metode secara khusus yang digunakan untuk transformasi data. Yang perlu diperhatikan adalah, ketika beberapa variabel independen memiliki range data yang berbeda, Bapak dapat menerapkan transformasi data pada setiap variabel independen yanga ada. 
Namun apabila Bapak ingin mengetahui perbedaan dari masing-masing transformasi data, berikut referensi yang bisa Bapak baca https://towardsdatascience.com/types-of-transformations-for-better-normal-distribution-61c22668d3b9 Pada artikel tersebut dijelaskan bagaimana cara untuk melakukan transformasi data untuk mengubah data sedikit mendekati distribusi normal. 

Demikian jawaban dari saya, semoga cukup membantu :)
